src/facilities/facilities.service.ts

```diff
@@ -93,15 +98,103 @@ export class FacilitiesService {
     return this.facilityMap.one(facility);
   }
 
-  async getAllFacilityAttributes(
+  async streamFacilitiesUnitAttributes(
+    req: Request,
     facilityAttributesParamsDTO: FacilityAttributesParamsDTO,
+  ): Promise<StreamableFile> {
+    const stream = await this.facilityUnitAttributesRepository.streamAllFacilityUnitAttributes(
+      facilityAttributesParamsDTO,
+    );
+    req.res.setHeader(
+      'X-Field-Mappings',
+      JSON.stringify(fieldMappings.facilities.attributes),
+    );
+    const toDto = new Transform({
+      objectMode:true,
+      transform(data, _enc, callback) {
+        delete data.id;
+
+        const commercialOperationDate = new Date(data.commercialOperationDate);
+        data.commercialOperationDate = commercialOperationDate.toISOString().split('T')[0];
+
+        let associatedGeneratorsAndNameplateCapacityStr = '';
+        const array = [data.ownDisplay, data.oprDisplay];
+        const ownOprList = array
+          .filter(e => e)
+          .join(',')
+          .slice(0, -1)
+          .split('),');
+        const ownOprUniqueList = [...new Set(ownOprList)];
+        const ownerOperator = ownOprUniqueList.join('),');
+
+        const generatorIdArr = data.associatedGeneratorsAndNamePlateCapacity?.split(', ');
```

Comment:
make the p lowercase in Nameplate 

const generatorIdArr = data.associatedGeneratorsAndNameplateCapacity?.split(', ');